apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx

```diff
@@ -5,11 +5,33 @@ import {borderRadius} from '@cdo/apps/lib/levelbuilder/constants';
 import EvidenceDescriptions from './EvidenceDescriptions';
 import Button from '../../../templates/Button';
 
-export default function LearningGoalItem({deleteItem}) {
-  const [aiEnabled, setAiEnabled] = useState(false);
+export default function LearningGoalItem({
+  deleteItem,
+  exisitingLearningGoalData,
+  handleAiEnabledChange,
+  handleLearningGoalNameChange,
+}) {
+  const [updatedAiEnabled, setUpdatedAiEnabled] = useState(
+    exisitingLearningGoalData.aiEnabled
+  );
+
+  const [updatedKeyConcept, setUpdatedKeyConcept] = useState(
+    exisitingLearningGoalData.learningGoal
+  );
+
+  const [learningGoalData] = useState(exisitingLearningGoalData);
 
   const handleCheckboxChange = () => {
-    setAiEnabled(!aiEnabled);
+    setUpdatedAiEnabled(!updatedAiEnabled);
+    handleAiEnabledChange(learningGoalData.id);
+  };
+
+  const handleKeyConceptChange = event => {
```

Comment:
In order to pass data to and from all the components, I have added these handlers throughout the files.  Open to better/cleaner ways to accomplish this goal

Reply:
I'm curious if `updateAiEnabled` and `updatedKeyConcept` are needed or if you can just use the corresponding fields in  `existingLearningGoalData`

Reply:
Sorry, I'm missing the change you made. Can you point to the commit?

Reply:
[This](https://github.com/code-dot-org/code-dot-org/pull/53432/commits/f3416c5b1e1f9fe2ecd8bc9a5a4eff3a366894f4) is the commit where I made the change.  I ended up creating a `updateLearningGoal` function that would modify a learning goal's data and then using that function inside the handler.

Reply:
gotcha! Do you still need `updatedAiEnabled` or can you just use `exisitingLearningGoalData.aiEnabled` (same thing for key concept)

Reply:
Thanks all.  I will be making this change to just rely on `exisitingLearningGoalData.aiEnabled` (and `exisitingLearningGoalData.learningGoal`) instead of duplicating it in state.  I had a misconception that these couldn't be passed through as props.